specification/0041-hashed-mappings-specification.md

```diff
@@ -0,0 +1,143 @@
+# Hashed Mapping Specification
+
+## Summary
+
+This document intends to revise and formalize the process for creating a
+hashed mapping as originally proposed in the hashed mojmap [RFC 19](https://github.com/QuiltMC/rfcs/blob/master/rfc/0019-hashed-mojmap.md).
+
+## Motivation
+
+Hashed Mojmap is intended to replace intermediary as our runtime mappings.
+It provides some benefits over intermediary, most notably the reduced maintenance cost.
+
+One of the design goals of hashed was, that it is not built incrementally,
+but can always be creating from an existing official mapping set without knowledge of earlier hashed versions.
+This also means that it should be possible for anyone to generate hashed mappings,
+even if they don't plan to use the mappings-hasher designed by Quilt.
+In order to allow this, a specification for the hashed mapping is needed.
+
+In order to specify our intended behaviour, this document describes the process to generate a hashed mapping.
+Anything not specified in this RFC is considered a bug in the generation of hashed mappings.
+
+## Explanation
+
+Given an original mappings set and the corresponding classes, a new mapping set is created with the following properties:
+- All new mappings are a hash of the corresponding original mapping.
+- All new mappings are unique in the new mapping set where possible.
+- The new mappings must ensure that the resulting remapped jar can be verified and executed.
+- The new mapping set should be as resilient to changes in the original mapping set and corresponding classes as possible.
+
+In order to achieve this, the following operations are performed on the original mapping set:
+- Convert all class/method/field names into their corresponding "raw" name.
+- Ensure methods that are required to have the same name do so.
+- Hash all raw names using the same algorithm.
+- Mark all class/method/field names with a corresponding prefix.
+- Prefix hashed class mappings with a default package.
+- Strip any mapping information that is unnecessary.
+
+### Raw Names
+For classes, the raw name is their binary name with the package omitted where possible.
```

Comment:
In what circumstances is the package omitted from the raw name of a class? While the examples do give some indication of this, it would be better for this to be specified directly.

Reply:
Thanks, I lost that part during reformatting!